### DISCUSSION TITLE
Using Github App how to get private organisations

### DISCUSSION THREAD

**Discussion by Sreejith-Websphere** [community/community#77398] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I am using Github App where after github app authorization i need to list private organisations along with public organisations
using users/{userName}/orgs i can only see public organisations

i played around with permissions&events -> organisation permissions but still it is not listing private orgs
and i checked /user/orgs (saw it works with oAuth Apps) was returning empty when accessing using github app

Note: we had a requirement where i need to list all the organisations(public/private) using github app for the user and need to compare with users existing installations

**Comment by b8zeek** [comment]
If I understood the question, you want to list the organizations of the users using the app?

If that's the case, you must grant the necessary permissions to the GitHub App. Follow the steps to ensure the correct permissions are set:

When creating or editing your GitHub App, under the `Permissions & events` section, set the `Organization permissions` to `Read-only` or `Read & write` depending on what you want to do.

Under the `Where can this GitHub App be installed?` section, ensure that `Any account` is selected to allow the app to be installed on user and organization accounts.

Save the changes. ✅

Now, when these permissions are set, the users who install your app and authorize it will allow the app to access their public and private organizations.

Hope this is what you're looking for. Cheers. 🍺 

🍀

**Reply by Sreejith-Websphere** [reply]
Thanks @bejzik8 for the answer

but we are looking on the reverse way, please correct me will it works
our requirement is
1. Authorize using Github app
2. Using Access token list organisations (public and private), i already given Organization permissions in  Permissions & events section but not listing the private organisations(this is where i am struck)
3. /users/{username}/orgs - lists only public organisations
4. /user/orgs - response is empty
5. After listing organisations i need to achieve like below screenshot (if organisation is not installed yet)

<img width="821" alt="Screenshot 2023-11-30 at 4 32 05 PM" src="https://github.com/community/community/assets/87368885/f46bc92e-5820-4652-bb08-ae2112827824">

**Reply by b8zeek** [reply]
No worries buddy. ☺️

So basically you want to use the GitHub's API to read the user's private data? There's definitely no way to achieve that!

You would have to use the flow I described previously.

Or maybe you can share the desired outcome and we can try to help.

Anyway, good luck. 🍺 

🍀

**Reply by Sreejith-Websphere** [reply]
one more doubt if we need to fetch all of the organization memberships (public and private) for the authenticated user, use the [List organizations for the authenticated user](https://docs.github.com/rest/orgs/orgs#list-organizations-for-the-authenticated-user) API instead using OAuth scope (read:org), why cant we achieve the same using Github apps authorization access token as we already giving permissions (permissions&events -> organisation permissions) while creating the github app

**Reply by b8zeek** [reply, marked as answer]
Because GitHub Apps and OAuth Apps have different scopes and permissions; GitHub Apps have more granular access controls compared to OAuth Apps and the reason is simple... OAuth Apps act on behalf of a user while GitHub Apps act on behalf of themselves.

In your case, there isn't a direct equivalent to the `read:org` scope in OAuth Apps.

Hope this clarifies it buddy; GL. 🍺 

🍀

**Reply by Sreejith-Websphere** [reply]
Thanks @bejzik8 for the clarification

**Reply by b8zeek** [reply]
No worries bro.

🍀
